rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/SSHInfrastructureV2.java

```diff
@@ -199,24 +200,37 @@ public void startNodeImpl(final HostTracker hostTracker, final int nbNodes, fina
         } catch (KeyException e) {
             throw new RMException("Could not get base64 credentials", e);
         }
-        clb.setCredentialsValueAndNullOthers(credString);
 
         // add an expected node. every unexpected node will be discarded
         String cmdLine;
         String obfuscatedCmdLine;
-        try {
-            cmdLine = clb.buildCommandLine(true);
-            obfuscatedCmdLine = clb.buildCommandLine(false);
-        } catch (IOException e) {
-            throw new RMException("Cannot build the " + RMNodeStarter.class.getSimpleName() + "'s command line.", e);
-        }
+
+        cmdLine = initScriptGenerator.fillInAgentScriptProperties(this.startupScript,
```

Comment:
I believe you should add try catch to cmdLine and obfuscatedCmdLine

Reply:
the command line builder used to throw an exception however with the new class no exception handling is required